### ISSUE TITLE
Bug: action without being used in any transition is lost in the result alphabet

### ISSUE THREAD

**Issue by trinnguyen** [uniba-swt/ia-toolset#3]
## Steps
- Create a proc with an actions that has no related transitions, example proc Q:

```
actions { a, b }

proc P {
    act { a? }
    a?
}

proc Q {
    act { a!, b? }
    a!
}

init {
    sys p = P()
    sys q = Q()
    sys prod = product(p, q)
    
    sys q2 = q
    sys q3 = prune(q)
    sys q4 = restrict(q, a)
}
```
- In result process instances: `prod`, `q2`, `q3`, `q4`, the action `b` should remains in the alphabet, even though there is no related transitions:
  + Actual (bug): no input action in `prod`, `q2`, `q3`, `q4`
  + Expected: action `b?` presents in `prod`, `q2`, `q3`, `q4`

- Screenshot (debugger) that has the bug
<img width="1088" alt="Screenshot 2021-07-16 at 12 30 15" src="https://user-images.githubusercontent.com/5253933/125935032-a8b22f6b-f3fd-427d-80a2-e591a8c58845.png">

## Solution
- Affected classes
  + `MemProductOperation`
  + `MemCloner`
  + `MemPruner`
  + `MemRestrictOperation`
+ Correct implementation that reserve the action when transforming the source code to IAM:
  + `IamGraph`

**Comment by eyip002** [comment]
The output `a` should also be preserved for `q2`, `q3`, `q4`

**Comment by eyip002** [comment]
Fixed by commit de06166afaafd2dc5e5612ceb2387735d0bc32c6
